go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/karmada-io/karmada
 
-go 1.21.10
+go 1.22.3 // keep in sync with .go-version
```

Comment:
1.22.4 is live now,how about  update version to 1.22.4, it's resolve a security issue (CVE-2024-24790 ).

Releated:
- https://github.com/golang/go/issues/67680
- https://go.dev/doc/devel/release#go1.22.4

Reply:
+1